packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -465,6 +529,17 @@ private void finishWithSuccess(String imagePath) {
     clearMethodCallAndResult();
   }
 
+  private void finishWithListSuccess(ArrayList<String> imagePaths) {
+    if (pendingResult == null) {
+      for (String imagePath : imagePaths) {
+        cache.saveResult(imagePath, null, null);
```

Comment:
only one image can be saved in the "cache", so only the last one saved is in the cache, it is not what we want right?
We might need a different cache for list Images, or we can deprecate the old cache, and the new cache always have a list of images (when there is only one image, the list length is 1)

Reply:
I removed the `finishWithListSuccess` method since we can save only one image path. I think we can use the `finishWithSuccess` method for now. 
Maybe we can create another PR to update the cache.

Reply:
Sure, please create an issue and link a TODO to it. We also need to mentioned this "known issue" in the CHANGELOG, which also can be linked to the same issue.